### DISCUSSION TITLE
性能优化

### DISCUSSION THREAD

**Discussion by xiaoL1211** [apache/doris#19642] [A - General / Q&A]
Query:
    Summary:
          -  Start  Time:  2023-05-15  15:38:56
          -  End  Time:  2023-05-15  15:39:03
          -  Total:  6s867ms
          -  Query  State:  EOF
          -  Job  ID:  N/A
          -  Query  ID:  137762dd0d4f4aab-a5ec3ff7458685ec
          -  Query  Type:  Query
          -  Doris  Version:  doris-1.2.4-1
          -  User:  root
          -  Default  Db:  default_cluster:ent
          -  Sql  Statement:  SELECT  version,BITMAP_UNION_COUNT(count)  as  total  FROM  `version_hour_uv`  WHERE  (os_type  =  "windows")  and  ((day_time  >=  "2023-05-08  00:00:00  +0800  CST")  and  (day_time  <  "2023-05-16  00:00:00  +0800  CST"))  GROUP  BY  `version`
          -  Is  Cached:  No
          -  Total  Instances  Num:  3
          -  Instances  Num  Per  BE:  {100.223.63.188:8060=3}
          -  Trace  ID:  
        Execution  Summary:
              -  Analysis  Time:  1.814ms
              -  Plan  Time:  6.858ms
              -  Schedule  Time:  5.412ms
              -  Fetch  Result  Time:  6s850ms
              -  Write  Result  Time:  70.768us
              -  Wait  and  Fetch  Result  Time:  6s850ms
    Execution  Profile  137762dd0d4f4aab-a5ec3ff7458685ec:(Active:  6s866ms,  %  non-child:  100.00%)
        Fragment  0:
            Instance  137762dd0d4f4aab-a5ec3ff7458685ef  (host=TNetworkAddress(hostname:100.223.63.188,  port:9060)):(Active:  6s849ms,  %  non-child:  0.00%)
                  -  DoriBeVersion:  Unknown
                  -  FragmentCpuTime:  388.68us
                  -  RowsProduced:  4
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                VDataBufferSender  (dst_fragment_instance_id=137762dd0d4f4aab--5a13c008ba797a11):
                      -  AppendBatchTime:  280.148us
                          -  ResultRendTime:  260.899us
                          -  TupleConvertTime:  3.734us
                      -  NumSentRows:  4
                VEXCHANGE_NODE  (id=4):(Active:  6s848ms,  %  non-child:  99.74%)
                      -  BytesReceived:  0.00  
                      -  DataArrivalWaitTime:  6s848ms
                      -  DecompressBytes:  0.00  
                      -  DecompressTime:  0ns
                      -  DeserializeRowBatchTimer:  0ns
                      -  FirstBatchArrivalWaitTime:  6s848ms
                      -  PeakMemoryUsage:  2.30  KB
                      -  ProjectionTime:  0ns
                      -  RowsReturned:  4
                      -  RowsReturnedRate:  0
                      -  SendersBlockedTotalTimer(*):  0ns
        Fragment  1:
            Instance  137762dd0d4f4aab-a5ec3ff7458685ee  (host=TNetworkAddress(hostname:100.223.63.188,  port:9060)):(Active:  6s850ms,  %  non-child:  0.02%)
                  -  DoriBeVersion:  Unknown
                  -  FragmentCpuTime:  1.508ms
                  -  RowsProduced:  4
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                VDataStreamSender  (dst_id=4,  dst_fragments=[{"137762dd0d4f4aab-a5ec3ff7458685ef"}]):(Active:  79.866us,  %  non-child:  0.00%)
                      -  BlocksSent:  2
                      -  BrpcSendTime:  77.376us
                      -  BrpcSendTime.Wait:  1.447ms
                      -  BytesSent:  0.00  
                      -  CompressTime:  0ns
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  102.00  B
                      -  LocalSendTime:  35.83us
                      -  LocalSentRows:  4
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  3.13  KB
                      -  SerializeBatchTime:  0ns
                      -  SplitBlockDistributeByChannelTime:  0ns
                      -  SplitBlockHashComputeTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                VAGGREGATION_NODE  (id=3):(Active:  6s848ms,  %  non-child:  0.02%)
                      -  BuildTime:  0ns
                      -  DeserializeDataTime:  878.736us
                      -  ExecTime:  0ns
                      -  ExprTime:  0ns
                      -  GetResultsTime:  15.231us
                      -  HashTableComputeTime:  17.338us
                      -  HashTableInputCount:  4
                      -  HashTableIterateTime:  180ns
                      -  HashTableSize:  0
                      -  InsertKeysToColumnTime:  4.283us
                      -  MaxRowSizeInBytes:  0
                      -  MergeTime:  1.260ms
                      -  PeakMemoryUsage:  0.00  
                      -  ProjectionTime:  0ns
                      -  RowsReturned:  4
                      -  RowsReturnedRate:  0
                      -  SerializeDataTime:  0ns
                      -  SerializeKeyTime:  0ns
                      -  SerializeResultTime:  0ns
                      -  StreamingAggTime:  0ns
                    VEXCHANGE_NODE  (id=2):(Active:  6s847ms,  %  non-child:  99.72%)
                          -  BytesReceived:  0.00  
                          -  DataArrivalWaitTime:  6s847ms
                          -  DecompressBytes:  0.00  
                          -  DecompressTime:  0ns
                          -  DeserializeRowBatchTimer:  0ns
                          -  FirstBatchArrivalWaitTime:  6s847ms
                          -  PeakMemoryUsage:  2.13  KB
                          -  ProjectionTime:  0ns
                          -  RowsReturned:  4
                          -  RowsReturnedRate:  0
                          -  SendersBlockedTotalTimer(*):  0ns
        Fragment  2:
            Instance  137762dd0d4f4aab-a5ec3ff7458685ed  (host=TNetworkAddress(hostname:100.223.63.188,  port:9060)):(Active:  6s847ms,  %  non-child:  0.01%)
                  -  DoriBeVersion:  Unknown
                  -  FragmentCpuTime:  14.829ms
                  -  RowsProduced:  4
                BlockMgr:
                      -  BlockWritesOutstanding:  0
                      -  BlocksCreated:  0
                      -  BlocksRecycled:  0
                      -  BufferedPins:  0
                      -  BytesWritten:  0.00  
                      -  MaxBlockSize:  8.00  MB
                      -  TotalBufferWaitTime:  0ns
                      -  TotalEncryptionTime:  0ns
                      -  TotalIntegrityCheckTime:  0ns
                      -  TotalReadBlockTime:  0ns
                VDataStreamSender  (dst_id=2,  dst_fragments=[{"137762dd0d4f4aab-a5ec3ff7458685ee"}]):(Active:  405.409us,  %  non-child:  0.01%)
                      -  BlocksSent:  2
                      -  BrpcSendTime:  95.599us
                      -  BrpcSendTime.Wait:  237.850us
                      -  BytesSent:  0.00  
                      -  CompressTime:  0ns
                      -  IgnoreRows:  0
                      -  LocalBytesSent:  552.99  KB
                      -  LocalSendTime:  378.857us
                      -  LocalSentRows:  4
                      -  OverallThroughput:  0.0  /sec
                      -  PeakMemoryUsage:  3.19  KB
                      -  SerializeBatchTime:  0ns
                      -  SplitBlockDistributeByChannelTime:  0ns
                      -  SplitBlockHashComputeTime:  0ns
                      -  UncompressedRowBatchSize:  0.00  
                VAGGREGATION_NODE  (id=1):(Active:  6s846ms,  %  non-child:  0.10%)
                      -  ExecOption:  Streaming  Preaggregation
                      -  BuildTime:  4.635ms
                      -  DeserializeDataTime:  0ns
                      -  ExecTime:  4.198ms
                      -  ExprTime:  95.982us
                      -  GetResultsTime:  1.364ms
                      -  HashTableComputeTime:  104.711us
                      -  HashTableInputCount:  128
                      -  HashTableIterateTime:  478ns
                      -  HashTableSize:  0
                      -  InsertKeysToColumnTime:  5.124us
                      -  MaxRowSizeInBytes:  0
                      -  MergeTime:  0ns
                      -  PeakMemoryUsage:  0.00  
                      -  ProjectionTime:  0ns
                      -  RowsReturned:  4
                      -  RowsReturnedRate:  0
                      -  SerializeDataTime:  1.348ms
                      -  SerializeKeyTime:  0ns
                      -  SerializeResultTime:  1.368ms
                      -  StreamingAggTime:  0ns
                    VNewOlapScanNode(version_hour_uv)  (id=0):(Active:  6s839ms,  %  non-child:  99.61%)
                          -  PushDownPredicates:  []
                          -  KeyRanges:  ScanKeys:ScanKey=[windows,2023-05-08  00:00:00  :  windows,2023-05-16  00:00:00)
                          -  TabletIds:  [112868,  112854,  114354,  112260,  112556,  114362,  113452,  112568,  113168,  112256,  112252,  113162,  114358,  112852,  113458,  112560,  113464,  113152,  112270,  113166,  112570,  113154,  112566,  113164,  112862,  113468,  112564,  112254,  113462,  112562,  112266,  112268,  112858,  112860,  112866,  112864,  114360,  114364,  112258,  113160,  112552,  113466,  112264,  113170,  114370,  114368,  112856,  112558,  112870,  114352,  114356,  113158,  113470,  113454,  112262,  113460,  113156,  112554,  113456,  114366]
                          -  AcuireRuntimeFilterTime:  773ns
                          -  GetNextTime:  6s835ms
                          -  MaxScannerThreadNum:  12
                          -  NewlyCreateFreeBlocksNum:  0
                          -  NumDiskAccess:  1
                          -  NumScanners:  60
                          -  PeakMemoryUsage:  0.00  
                          -  PreAllocFreeBlocksNum:  60
                          -  ProcessConjunctTime:  77.510us
                          -  ProjectionTime:  0ns
                          -  RowsRead:  128
                          -  RowsReturned:  128
                          -  RowsReturnedRate:  18
                          -  ScannerBatchWaitTime:  6s835ms
                          -  ScannerCtxSchedCount:  60
                          -  ScannerSchedCount:  60
                          -  ScannerWorkerWaitTime:  17.677ms
                          -  TabletNum:  60
                          -  TotalReadThroughput:  0
                        VScanner:
                              -  BlockConvertTime:  0ns
                              -  BlockFetchTime:  1s932ms
                              -  ReaderInitTime:  0ns
                              -  RowsDelFiltered:  0
                              -  ScannerConvertBlockTime:  0ns
                              -  ScannerCpuTime:  52.787ms
                              -  ScannerFilterTime:  25.337us
                              -  ScannerGetBlockTime:  613.390us
                              -  ScannerInitTime:  1.411ms
                              -  ScannerPrefilterTime:  0ns
                            SegmentIterator:
                                  -  BitmapIndexFilterTimer:  13.638us
                                  -  BlockConditionsFilteredTime:  42.259us
                                  -  BlockInitSeekCount:  168
                                  -  BlockInitSeekTime:  1.29ms
                                  -  BlockInitTime:  1s203ms
                                  -  BlockLoadTime:  1s932ms
                                  -  BlocksLoad:  96
                                  -  CachedPagesNum:  0
                                  -  CompressedBytesRead:  2.82  MB
                                  -  DecompressorTimer:  1.963ms
                                  -  FirstReadSeekCount:  48
                                  -  FirstReadSeekTime:  676.246ms
                                  -  FirstReadTime:  727.240ms
                                  -  IOTimer:  1s915ms
                                  -  LazyReadSeekCount:  0
                                  -  LazyReadSeekTime:  0ns
                                  -  LazyReadTime:  0ns
                                  -  NumSegmentFiltered:  0
                                  -  NumSegmentTotal:  48
                                  -  OutputColumnTime:  15.711us
                                  -  RawRowsRead:  128
                                  -  RowsBitmapIndexFiltered:  0
                                  -  RowsBloomFilterFiltered:  0
                                  -  RowsConditionsFiltered:  0
                                  -  RowsKeyRangeFiltered:  0
                                  -  RowsShortCircuitPredFiltered:  0
                                  -  RowsShortCircuitPredInput:  0
                                  -  RowsStatsFiltered:  0
                                  -  RowsVectorPredFiltered:  0
                                  -  RowsVectorPredInput:  0
                                  -  ShortPredEvalTime:  0ns
                                  -  TotalPagesNum:  312
                                  -  UncompressedBytesRead:  3.33  MB
                                  -  VectorPredEvalTime:  0ns

如上： Fetch  Result  Time: 很耗时间，怎么优化。版本1.2.4.1

**Comment by yagagagaga** [comment]
According to your profile, we can see that Doris read about 128 Rows and cost 6 sec in `VNewOlapScanNode`. This is really abnormal.   You can try again a few more times or try restarting the server and retry. If it still doesn't work, then this may be a bug. Can you provide me with your table DDL and data, and I will help you solve the problem.

**Reply by xiaoL1211** [reply]
How much is your WeChat account?

**Reply by yagagagaga** [reply]
scan the qr-code

![](https://user-images.githubusercontent.com/21151126/222130177-a85c1e2b-bdaa-4c96-98bc-2c4c8da02615.JPG)

**Reply by xiaoL1211** [reply]
![50dc58ab5023de0b398c7b3476041aa](https://github.com/apache/doris/assets/5170688/dba872e6-e1fb-4d26-960c-26e786717081)
![280e817bcff34277bc6b08fbda8211b](https://github.com/apache/doris/assets/5170688/966e6616-9861-4d4e-972c-77675302f8fe)




What is the reason for this

**Reply by xiaoL1211** [reply]
What is the reason for this
